src/org/centos/pipeline/PipelineUtils.groovy

```diff
@@ -898,24 +898,51 @@ def watchForMessages(String msg_provider, String message) {
 
     def messageVars = getVariablesFromMessage(message)
 
-    topicsToWatchFor = ["org.centos.stage.ci.pipeline.package.running",
-                        "org.centos.stage.ci.pipeline.package.complete",
-                        "org.centos.stage.ci.pipeline.compose.running",
-                        "org.centos.stage.ci.pipeline.compose.complete",
-                        "org.centos.stage.ci.pipeline.compose.test.integration.queued",
-                        "org.centos.stage.ci.pipeline.compose.test.integration.running",
-                        "org.centos.stage.ci.pipeline.compose.test.integration.complete",
-                        "org.centos.stage.ci.pipeline.complete"]
-
-    topicsToWatchFor.each {
-        echo "Waiting for topic : " + it
+    messageContentValidationMap = [:]
+    messageContentValidationMap['org.centos.stage.ci.pipeline.package.running'] =
+            ["branch", "repo"]
+    messageContentValidationMap['org.centos.stage.ci.pipeline.package.complete'] =
+            ["branch", "repo"]
```

Comment:
iiuc, these are the fields that we want to make sure are not empty in the message? If so, can we add original_spec_nvr to the list for package.complete? Just because that one was specifically requested to be added

Reply:
exactly. will do. but we can review the complete list once the tests pass...